lang/wasm_alien_thread_runner.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <exception>
+#include <seastar/core/alien.hh>
+#include <seastar/core/reactor.hh>
+
+#include "lang/wasm.hh"
+#include "lang/wasm_alien_thread_runner.hh"
+
+using namespace std::chrono_literals;
+
+namespace wasm {
+
+std::optional<wasm_compile_task> task_queue::pop_front() {
+    std::unique_lock<std::mutex> lock(_mut);
+    _cv.wait(lock, [this] { return !_pending.empty(); });
+    std::optional<wasm_compile_task> work_item = std::move(_pending.front());
+    _pending.pop();
+    return work_item;
+}
+
+void task_queue::push_back(std::optional<wasm_compile_task> work_item) {
+    std::unique_lock<std::mutex> lock(_mut);
+    if (work_item && _pending.size() >= QUEUE_SIZE) {
+        seastar::alien::run_on(seastar::engine().alien(), work_item->shard, [&done = work_item->done] {
+            done.set_exception(std::runtime_error("alien_thread_runner queue is full"));
```

Comment:
Could this cause use to fail to load?

Consider 150 shards trying to compile a UDF in parallel. The last 30 will fail. How do we recover from this?

Better to have an unbounded queue.

Reply:
Ok